src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
+        T returnItem = currentNode.item;
+        currentNode.item = value;
+        return returnItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        verifyIndex(index);
+        if (index == 0) {
+            return removeFirstNode();
+        }
+        iterateCurrentNode(index);
+        assignPreviousAndNextNode();
+        unlink(currentNode);
+        size--;
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if (size == 0) {
+            throw new NullPointerException("The list is empty");
+        }
+        if (object == null ? head.item == null : object.equals(head.item)) {
+            return removeFromTheHead();
+        }
+        findInTheMiddle(object);
+        if (currentNode == null) {
+            return false;
+        }
+        assignPreviousAndNextNode();
+        unlink(currentNode);
+        size--;
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void iterateCurrentNode(int index) {
+        currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+    }
+
+    private void verifyIndex(int index) {
+        if (index < 0 || index > size - 1) {
+            throw new IndexOutOfBoundsException("The index passed must be equal "
+                    + "or greater than 0 and less than " + size);
+        }
+    }
+
+    private void addNodeInTheMiddle(T value) {
+        previousNode = currentNode.prev;
+        Node<T> newElement = new Node<>(value);
+        newElement.prev = previousNode;
+        newElement.next = currentNode;
+        if (currentNode == head || currentNode != null && currentNode.equals(head)) {
+            head = newElement;
+        }
+        if (previousNode != null) {
+            previousNode.next = newElement;
+        }
+        currentNode.prev = newElement;
+        size++;
+    }
+
+    private T removeFirstNode() {
+        final Node<T> removedElement = head;
+        head = head.next;
+        if (head != null) {
+            head.prev = null;
+        }
+        size--;
+        return removedElement.item;
+    }
+
+    private void assignPreviousAndNextNode() {
+        previousNode = currentNode.prev;
+        nextNode = currentNode.next;
+        if (nextNode != null) {
+            nextNode.prev = previousNode;
+        }
+        if (previousNode != null) {
+            previousNode.next = nextNode;
+        }
+    }
+
+    private void unlink(Node<T> node) {
+        node.prev = null;
+        node.next = null;
+        if (tail.equals(node)) {
+            Node<T> newNode = tail;
+            newNode.next = null;
+            newNode.prev = null;
+        }
+        if (head.equals(node)) {
```

Comment:
The logic inside the 'unlink' method for handling the head node is incorrect. Ensure that the head is updated correctly when the node is the head.